src/aspire/source/simulation.py

```diff
@@ -413,7 +413,7 @@ def eigs(self):
         C = self.C
         vols_c = self.vols - self.mean_true()
 
-        p = np.ones(C) / C
+        p = np.ones(C, dtype=self.dtype) / C.astype(self.dtype)
```

Comment:
since C should be an integer, maybe better to use `np.full` and just make the thing we want here.